### PR TITLE
chore: Quarkus 1.13 migration

### PR DESCRIPTION
**Pull request by aureamunoz**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -141,18 +141,21 @@
...
+                <plugin>
+                    <groupId>io.quarkus</groupId>
+                    <artifactId>quarkus-maven-plugin</artifactId>
+                    <version>${quarkus-plugin.version}</version>
+                    <extensions>true</extensions>
+                    <executions>
+                        <execution>
+                            <goals>
+                                <goal>build</goal>
+                                <goal>generate-code</goal>
```
Comment (metacosm) [inline pom.xml:153]: Why do we need these 2 new goals?

File: src/main/docker/Dockerfile.jvm [hunk 2]
```diff
@@ -21,34 +21,36 @@
...
+FROM registry.access.redhat.com/ubi8/ubi-minimal:8.3
 
 ARG JAVA_PACKAGE=java-11-openjdk-headless
 ARG RUN_JAVA_VERSION=1.3.8
 
 ENV LANG='en_US.UTF-8' LANGUAGE='en_US:en'
 
 # Install java and the run-java script
-# Also set up permissions for user `1000`
+# Also set up permissions for user `1001`
```
Comment (metacosm) [inline src/main/docker/Dockerfile.jvm:32]: Why did you change the userid?

File: src/main/docker/Dockerfile.legacy-jar [hunk 3]
```diff
@@ -21,33 +21,34 @@
...
+COPY target/lib/* /deployments/lib/
+COPY target/*-runner.jar /deployments/app.jar
 
 EXPOSE 8080
-USER 1001
+USER 1000
 
-ENTRYPOINT [ "/deployments/run-java.sh" ]
\ No newline at end of file
+#ENTRYPOINT [ "/deployments/run-java.sh" ]
```
Comment (metacosm) [inline src/main/docker/Dockerfile.legacy-jar:54]: There's no entry point anymore?

File: test.sh [hunk 4]
```diff
@@ -0,0 +1,15 @@
+ISSUES_MANAGER_VERSION=$(xpath -q -e  '/project/version/text()' pom.xml)
```
Comment (metacosm) [inline test.sh:1]: Is this file still relevant? It uses the old name and the old-style jars…

**Review by metacosm — COMMENTED** [review]
I have some questions…

File: src/main/docker/Dockerfile.jvm (hunk 2 above)
Comment (cmoulliard) [reply src/main/docker/Dockerfile.jvm:32]: User UID should be `1000` as needed by jenkins jnlp agent - see here for more information: https://github.com/jenkinsci/kubernetes-plugin#pipeline-sh-step-hangs-when-multiple-containers-are-used

File: src/main/docker/Dockerfile.legacy-jar (hunk 3 above)
Comment (cmoulliard) [reply src/main/docker/Dockerfile.legacy-jar:54]: I will create a separate Dockerfile to be used to build the image when used with jenkins as some changes like this or UID = 1000 are needed

File: test.sh (hunk 4 above)
Comment (cmoulliard) [reply test.sh:1]: If possible, we should remove or replace it like also to use xpath which is not a tool installed on all the OS.

File: pom.xml (hunk 1 above)
Comment (aureamunoz) [reply pom.xml:153]: https://github.com/quarkusio/quarkus/wiki/Migration-Guide-1.12#quarkus-maven-plugin

File: test.sh (hunk 4 above)
Comment (aureamunoz) [reply test.sh:1]: oups!! no, no, I pushed it by mistake. I will remove it

File: src/main/docker/Dockerfile.jvm (hunk 2 above)
Comment (aureamunoz) [reply src/main/docker/Dockerfile.jvm:32]: I didn't actually change the user, this `Dockerfile.jvm` is the old `fast-jar` file that has became the default Dockerfile since 1.12 Quarkus version, so Dockerfile.jvm. It contained already the user `1001` and as I didnt know that the `1000` user was needed, I kept the `1001`. I will put the `1001` everywhere.

File: src/main/docker/Dockerfile.legacy-jar (hunk 3 above)
Comment (aureamunoz) [reply src/main/docker/Dockerfile.legacy-jar:54]: We can use the `Dockerfile.jvm` for using with Jenkins, I think we are not using it for other purpose. There is also the `legacy-jar` Dockerfile (old Dockerfile.jvm prior to 1.12 version of Quarkus)

File: pom.xml (hunk 1 above)
Comment (cmoulliard) [reply pom.xml:153]: Should we also add `<goal>generate-code-tests</goal>` ?

File: src/main/docker/Dockerfile.jvm [hunk 5]
```diff
@@ -21,34 +21,36 @@
 # docker run -i --rm -p 8080:8080 -p 5005:5005 -e JAVA_ENABLE_DEBUG="true" quarkus/release-manager-jvm
 #
 ###
-FROM registry.access.redhat.com/ubi8/ubi-minimal:8.1
+FROM registry.access.redhat.com/ubi8/ubi-minimal:8.3
```
Comment (cmoulliard) [inline src/main/docker/Dockerfile.jvm:24]: Should we mention the version instead of using latest ? Why: As the minimal image dont include JDK or maven tools, then we can go ahead and use latest to always get the updated installed packages (curl, bash, lib, ...) : https://catalog.redhat.com/software/containers/ubi8-minimal/5c64772edd19c77a158ea216?gti-tabs=unauthenticated&container-tabs=packages

File: src/main/docker/Dockerfile.jvm (hunk 2 above)
Comment (cmoulliard) [reply src/main/docker/Dockerfile.jvm:32]: +1

File: src/main/docker/Dockerfile.jvm [hunk 6]
```diff
@@ -21,34 +21,36 @@
...
+COPY --chown=1001 target/quarkus-app/lib/ /deployments/lib/
+COPY --chown=1001 target/quarkus-app/*.jar /deployments/
+COPY --chown=1001 target/quarkus-app/app/ /deployments/app/
+COPY --chown=1001 target/quarkus-app/quarkus/ /deployments/quarkus/
 
 EXPOSE 8080
-USER 1000
+USER 1001
 
 #ENTRYPOINT [ "/deployments/run-java.sh" ]
```
Comment (cmoulliard) [inline src/main/docker/Dockerfile.jvm:56]: Should be `ENTRYPOINT [ "/deployments/run-java.sh" ] ` as we will use a different Dockerfile to build the image for jenkins jnlp.

File: src/main/docker/Dockerfile.legacy-jar [hunk 7]
```diff
@@ -21,33 +21,34 @@
 # docker run -i --rm -p 8080:8080 -p 5005:5005 -e JAVA_ENABLE_DEBUG="true" quarkus/release-manager-jvm
 #
 ###
-FROM registry.access.redhat.com/ubi8/ubi-minimal:8.1
+FROM registry.access.redhat.com/ubi8/ubi-minimal:8.3
```
Comment (cmoulliard) [inline src/main/docker/Dockerfile.legacy-jar:24]: See previous remark version vs latest

File: src/main/docker/Dockerfile.native [hunk 8]
```diff
@@ -14,7 +14,7 @@
 # docker run -i --rm -p 8080:8080 quarkus/release-manager
 #
 ###
-FROM registry.access.redhat.com/ubi8/ubi-minimal:8.1
+FROM registry.access.redhat.com/ubi8/ubi-minimal:8.3
```
Comment (cmoulliard) [inline src/main/docker/Dockerfile.native:17]: See previous comment version vs latest

File: pom.xml (hunk 1 above)
Comment (aureamunoz) [reply pom.xml:153]: The `generate-code-tests` goal has been also added. Line 154

File: src/main/docker/Dockerfile.legacy-jar (hunk 3 above)
Comment (aureamunoz) [reply src/main/docker/Dockerfile.legacy-jar:54]: Ok, so, we don't need the `wget` for the certificates in these files, only in the Jenkins specific one, right? @cmoulliard

File: src/main/docker/Dockerfile.jvm (hunk 5 above)
Comment (aureamunoz) [reply src/main/docker/Dockerfile.jvm:24]: Ok, I don't have an opinion against it

File: src/main/docker/Dockerfile.jvm (hunk 6 above)
Comment (aureamunoz) [reply src/main/docker/Dockerfile.jvm:56]: yes, let's keep these files as generated initially

**Comment by cmoulliard** [comment]
LGTM. Can be merged if Chris agrees @aureamunoz

File: src/main/docker/Dockerfile.legacy-jar (hunk 3 above)
Comment (cmoulliard) [reply src/main/docker/Dockerfile.legacy-jar]: Yes. wget is not needed in this case. This is only needed for jenkins jnlp @aureamunoz

File: src/main/docker/Dockerfile.jvm (hunk 5 above)
Comment (cmoulliard) [reply src/main/docker/Dockerfile.jvm]: What do you then decide ? @aureamunoz

File: src/main/docker/Dockerfile.jvm (hunk 6 above)
Comment (cmoulliard) [reply src/main/docker/Dockerfile.jvm]: Can you then remove the `#` in front of the `ENTRYPOINT` ? @aureamunoz

File: src/main/docker/Dockerfile.jvm (hunk 5 above)
Comment (aureamunoz) [reply src/main/docker/Dockerfile.jvm]: `latest` version

File: src/main/docker/Dockerfile.jvm (hunk 6 above)
Comment (aureamunoz) [reply src/main/docker/Dockerfile.jvm]: yes, it's done in the lasts commits, isn't it?

File: test.sh (hunk 4 above)
Comment (metacosm) [reply test.sh:1]: This file needs to be removed.

**Review by metacosm — CHANGES_REQUESTED** [review]
Extra file needs to be removed.

File: test.sh (hunk 4 above)
Comment (aureamunoz) [reply test.sh]: :-S
Done!

**Comment by metacosm** [comment]
The github actions need to be updated as well, apparently.

**Review by metacosm — APPROVED** [review]
LGTM
